### PR TITLE
Add infinity to the evolution comparators

### PR DESCRIPTION
**Pull request by wthrowe**
This can be used to simplify #2983 and some upcoming dense output code.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: tests/Unit/Time/Test_EvolutionOrdering.cpp [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 #include "Framework/TestingFramework.hpp"
 
+#include <limits>
```
Comment (kidder) [inline tests/Unit/Time/Test_EvolutionOrdering.cpp]: alphabet (this is what clang-tidy is unhappy with)

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash the fix

**Review by kidder — DISMISSED** [review]

**Comment by kidder** [comment]
needs a rebase to include #3018 so CI passes

File: src/Time/EvolutionOrdering.hpp [hunk 2]
```diff
@@ -3,108 +3,83 @@
...
-
-  // NOLINTNEXTLINE(google-runtime-references)
-  void pup(PUP::er& p) noexcept { p | time_runs_forward; }
-};
-
-template <>
-struct evolution_less<void> {
-  bool time_runs_forward = true;
   template <typename T, typename U>
   constexpr auto operator()(T&& t, U&& u) const noexcept {
```
Comment (nilsvu) [inline src/Time/EvolutionOrdering.hpp]: I think it would be clearer to change `auto` -> `bool` here.

File: src/Time/EvolutionOrdering.hpp [hunk 3]
```diff
@@ -3,108 +3,83 @@
...
+///
+/// \see std::less
+template <typename T = void>
+using evolution_less = evolution_comparator<T, std::less>;
 
-  // NOLINTNEXTLINE(google-runtime-references)
-  void pup(PUP::er& p) noexcept { p | time_runs_forward; }
-};
+/// \ingroup TimeGroup
+/// \copydoc evolution_less
```
Comment (nilsvu) [inline src/Time/EvolutionOrdering.hpp:80]: Wrap these in `// @{` instead? Up to you.

**Review by nilsvu — CHANGES_REQUESTED** [review]

**Comment by nilsvu** [comment]
Please squash this directly while you're rebasing.

File: src/Time/EvolutionOrdering.hpp (hunk 3 above)
Comment (wthrowe) [reply src/Time/EvolutionOrdering.hpp:80]: Tried that.  Didn't work.

File: src/Time/EvolutionOrdering.hpp (hunk 2 above)
Comment (wthrowe) [reply src/Time/EvolutionOrdering.hpp]: It doesn't have to be bool.  Should actually probably be `decltype(auto)`, though.  I'll fix that. 
 https://en.cppreference.com/w/cpp/utility/functional/less_void

**Comment by wthrowe** [comment]
The last commit is new.

**Review by nilsvu — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**